SiriusAdapter replaced with SiriusExport and adaptations to AssayGeneratorMetabo

axelwalter opened the pull request:
## Description

SiriusExport should produce file required to run SIRIUS externally (.ms file) similar to GNPSExport. AssayGeneratorMetabo relies heavily on SIRIUS and therefore requires some rework to accept an existing SIRIUS workspace as input instead of calling SIRIUS during the workflow.

Draft PR with work in progress.

- [x] Documentation needs to be updated
- [x] SiriusExport pre-processing still required in AssayGeneratorMetabo (works well, but there might be better solution)
- [x] correct precursor charges need to be determined when using SIRIUS for fragment annotation

## Checklist
- [ ] Make sure that you are listed in the AUTHORS file
- [ ] Add relevant changes and new features to the CHANGELOG file
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] New and existing unit tests pass locally with my changes
- [ ] Updated or added python bindings for changed or new classes (Tick if no updates were necessary.)

### How can I get additional information on failed tests during CI
<details>
  <summary>Click to expand</summary>
If your PR is failing you can check out

- The details of the action statuses at the end of the PR or the "Checks" tab.
- http://cdash.openms.de/index.php?project=OpenMS and look for your PR. Use the "Show filters" capability on the top right to search for your PR number.
  If you click in the column that lists the failed tests you will get detailed error messages.

</details>

### Advanced commands (admins / reviewer only)
<details>
  <summary>Click to expand</summary>
  
- `/reformat` (experimental) applies the clang-format style changes as additional commit. Note: your branch must have a different name (e.g., yourrepo:feature/XYZ) than the receiving branch (e.g., OpenMS:develop). Otherwise, reformat fails to push.
- setting the label "NoJenkins" will skip tests for this PR on jenkins (saves resources e.g., on edits that do not affect tests)
- commenting with `rebuild jenkins` will retrigger Jenkins-based CI builds
  
</details>

---
:warning: Note: Once you opened a PR try to minimize the number of *pushes* to it as every push will trigger CI (automated builds and test) and is rather heavy on our infrastructure (e.g., if several pushes per day are performed).

axelwalter committed:
rework SiriusAdapter to SiriusExport

axelwalter committed:
AssayGeneratorMetabo rework
- does not execute SIRIUS directly
- takes SIRIUS workspace as input

axelwalter committed:
remove sirius executable param

axelwalter committed:
always set precursor charge to one

timosachsenberg commented on CHANGELOG:
```diff
@@ -385,7 +385,7 @@ Deprecated and removed tools:
   
 Changed Tools:
-- SiriusAdapter now supports several input data
+- SiriusExport now supports several input data
```
accidental rename (also below)

timosachsenberg commented on doc/OpenMS_tutorial/developer_Tutorial.doxygen:
```diff
@@ -270,7 +270,7 @@ Within the ANALYSIS folder, you can find several important tools
 
 
-For the sake of completeness you will find a short list of the THIRDPARTY tools, which are integrated via wrappers into the %OpenMS framework (usually called -Adapter e.g. SiriusAdapter)
+For the sake of completeness you will find a short list of the THIRDPARTY tools, which are integrated via wrappers into the %OpenMS framework (usually called -Adapter e.g. SiriusExport)
```
```suggestion
For the sake of completeness you will find a short list of the THIRDPARTY tools, which are integrated via wrappers into the %OpenMS framework (usually called -Adapter e.g. CometAdapter)
```

timosachsenberg commented on doc/doxygen/public/TOPP.doxygen:
```diff
@@ -199,7 +199,7 @@
   - @subpage TOPP_AccurateMassSearch - Finds potential HMDB IDs within the given mass error window.
   - @subpage TOPP_MetaboliteSpectralMatcher - Identifies small molecules from tandem MS spectra.
-  - @subpage TOPP_SiriusAdapter - De novo metabolite identification.
+  - @subpage TOPP_SiriusExport - De novo metabolite identification.
```
```suggestion
  - @subpage TOPP_SiriusExport - Export for Sirius de novo metabolite identification.
```

timosachsenberg commented on src/openms/include/OpenMS/CONCEPT/Constants.h:
```diff
@@ -423,7 +423,7 @@ namespace OpenMS
 
       /** @name User parameters in/from Sirius annotated mzMLs
-        *  E.g., when running SiriusAdapter with annotation output
+        *  E.g., when running SiriusExport with annotation output
```
check

axelwalter replied:
Updated the description.

timosachsenberg commented on src/openms/include/OpenMS/FORMAT/DATAACCESS/CsiFingerIdMzTabWriter.h:
```diff
@@ -17,7 +17,7 @@ namespace OpenMS
 
           /**
-          @brief Internal structure used in @ref TOPP_SiriusAdapter that is used
+          @brief Internal structure used in @ref TOPP_SiriusExport that is used
```
check... sound like import

axelwalter replied:
The Sirius MzTabWriter files have been removed, they can be written by Sirius externally using the `write-summaries` flag.

timosachsenberg commented on src/openms/include/OpenMS/FORMAT/DATAACCESS/SiriusMzTabWriter.h:
```diff
@@ -18,11 +18,11 @@ namespace OpenMS
 
     /**
-    @brief Internal structure used in @ref TOPP_SiriusAdapter that is used
+    @brief Internal structure used in @ref TOPP_SiriusExport that is used
```
is this still correct? or is this something now done by AssayGeneratorMetabo? also check comments/doc below

timosachsenberg commented on src/openms/source/ANALYSIS/ID/SiriusExportAlgorithm.cpp:
```diff
@@ -0,0 +1,257 @@
+      }
+    }
+
+    class OPENMS_DLLAPI SiriusWorkspaceIndex
+    {
+    public:
+      int array_index, scan_index;
+
+      SiriusWorkspaceIndex(int array_index, int scan_index) : array_index {array_index}, scan_index {scan_index} {}
+    };
+
```
```suggestion
    struct OPENMS_DLLAPI SiriusWorkspaceIndex
    {
      SiriusWorkspaceIndex(int array_index, int scan_index) : array_index {array_index}, scan_index {scan_index} {}
      int array_index, scan_index;
    };
```

timosachsenberg commented on src/openms/source/ANALYSIS/ID/SiriusExportAlgorithm.cpp:
```diff
@@ -0,0 +1,257 @@
+
+    bool SiriusExportAlgorithm::hasFullNameParameter(const OpenMS::String &name) const
+    {
+      for (auto it = param_.begin(); it != param_.end(); ++it)
+      {
+        if (it.getName() == name)
+        {
+          return true;
+        }
+      }
+      return false;
```
```suggestion
  return std::any_of(param_.begin(), param_.end(), [&](const auto& it) { return it.getName() == name; });
```

timosachsenberg commented on src/openms/source/FORMAT/DATAACCESS/SiriusFragmentAnnotation.cpp:
```diff
@@ -2,7 +2,7 @@
 //
 // --------------------------------------------------------------------------
-// $Maintainer: Oliver Alka $
+// $Maintainer: Oliver Alka., Axel Walter $
```
```suggestion
// $Maintainer: Oliver Alka, Axel Walter $
```

timosachsenberg commented on src/openms/source/FORMAT/DATAACCESS/SiriusFragmentAnnotation.cpp:
```diff
@@ -266,7 +266,7 @@ namespace OpenMS
         String adduct = sl[columnname_to_columnindex.at("adduct")];
         adduct.erase(std::remove_if(adduct.begin(), adduct.end(), ::isspace), adduct.end());
-        rank_filename.emplace(std::make_pair(sl[columnname_to_columnindex.at("rank")].toInt(),
+        rank_filename.emplace(std::make_pair(sl[columnname_to_columnindex.at("formulaRank")].toInt(),
```
did this change?

axelwalter replied:
Yes, with the latest SIRIUS the column is named formulaRank.

timosachsenberg commented on src/openms/source/FORMAT/MzTabM.cpp:
```diff
@@ -486,7 +486,7 @@ namespace OpenMS
         id_mslevel = 1;
       }
-      if (identification_software == "SiriusAdapter")
+      if (identification_software == "SiriusExport")
```
should this be backward compatible? e.g. use ||

timosachsenberg commented on src/topp/SiriusExport.cpp:
```diff
@@ -0,0 +1,170 @@
+
+  Sirius can be found on https://bio.informatik.uni-jena.de/software/sirius/ 
+
+  Please use Sirius Version 4.0.1
```
is this still what we use?

timosachsenberg commented on src/topp/AssayGeneratorMetabo.cpp:
```diff
@@ -23,9 +23,13 @@
+#include <QtCore/QProcess>
 #include <algorithm>
 #include <map>
+#include <boost/algorithm/string.hpp>
```
needed?

axelwalter replied:
Not needed any more. Parsing the compound.info file with CsvFile.

timosachsenberg commented on src/topp/AssayGeneratorMetabo.cpp:
```diff
@@ -433,74 +388,99 @@ class TOPPAssayGeneratorMetabo :
+          String file_path_compound_info = subdir + "/compound.info";
+          // open file and extract information
+          std::ifstream compound_info_file(file_path_compound_info);
+          if (compound_info_file.is_open())
```
could we use https://github.com/OpenMS/OpenMS/blob/develop/src/openms/include/OpenMS/FORMAT/CsvFile.h
? Mainly to reduce dependency on boost here...

axelwalter committed:
Update doc/doxygen/public/TOPP.doxygen

Co-authored-by: Timo Sachsenberg <timo.sachsenberg@uni-tuebingen.de>

axelwalter committed:
Update doc/OpenMS_tutorial/developer_Tutorial.doxygen

Co-authored-by: Timo Sachsenberg <timo.sachsenberg@uni-tuebingen.de>

axelwalter committed:
Update src/openms/source/ANALYSIS/ID/SiriusExportAlgorithm.cpp

Co-authored-by: Timo Sachsenberg <timo.sachsenberg@uni-tuebingen.de>

axelwalter committed:
Update src/openms/source/ANALYSIS/ID/SiriusExportAlgorithm.cpp

Co-authored-by: Timo Sachsenberg <timo.sachsenberg@uni-tuebingen.de>

axelwalter committed:
Update src/openms/source/FORMAT/DATAACCESS/SiriusFragmentAnnotation.cpp

Co-authored-by: Timo Sachsenberg <timo.sachsenberg@uni-tuebingen.de>

axelwalter committed:
fix accidental rename

axelwalter committed:
Remove SIRIUS and CSI MzTabWriter

axelwalter committed:
AssayGeneratorMetabo remove boost for string trimming

axelwalter committed:
use CsvFile to parse compound.info file

axelwalter committed:
edit description of SIRIUS params in Constants

axelwalter commented:
Thanks for the review! The comments are addressed in the recent commits. I will update the documentation next.

axelwalter committed:
remove unused functions

axelwalter committed:
update docs for SiriusExport

axelwalter committed:
update AssayGeneratorMetabo docs

axelwalter committed:
SiriusExport mandatory output

axelwalter committed:
update tests

axelwalter committed:
split AssayGeneratorMetabo tool
- new AssayGeneratorMetaboSirius tool

axelwalter committed:
SiriusExport
- export multiple mzML/featureXML files into one .ms file
- optional output: tsv file with CompoundInfo to use in AssayGeneratorMetabo

axelwalter committed:
remove unused imports

axelwalter committed:
read compound info table in AssayGeneratorMetaboSirius

axelwalter committed:
remove unused function to sort SIRIUS workspace

axelwalter committed:
export compound info only if they match a feature

axelwalter committed:
fix ambiguity groups with correct file index

axelwalter committed:
extract decoy specs from SIRIUS subdirs only if decoy_generation is set

axelwalter committed:
update resolveOverlappingTargetDecoyMassesByIndividualMassShift with mz tolerance

axelwalter committed:
rename function

axelwalter committed:
update pxd files

axelwalter committed:
update changelog

axelwalter committed:
update tests

axelwalter committed:
resolve conflicts and update SiriusFragementAnnotation_test

axelwalter committed:
fix test names

axelwalter committed:
set test dependencies

timosachsenberg commented on CHANGELOG:
```diff
@@ -18,8 +18,12 @@ PR - Pull Request (on GitHub), i.e. integration of a new feature or bugfix
+
 Library:
 - Extend FileHandler to support load and store operations for our major datastructures (spectra, features, identifications, etc.). Replaced file type specific code with the more generic FileHandler calls to decouple the IO code from other parts of the library.
+- better documentation for all SpectraFilter... tools (#7183) 
```
```suggestion
```

timosachsenberg commented on src/openms/include/OpenMS/ANALYSIS/TARGETED/MetaboTargetedTargetDecoy.h:
```diff
@@ -52,9 +52,10 @@ namespace OpenMS
     @param mass_to_add (e.g. CH2)
-
+    @param mz_tol m/z tolerarance for target and decoy transition masses to be considered overlapping
+    @param mz_tol_unit m/z tolerance unit
```
if you use a string then it should be documented what the possible options are

timosachsenberg commented on src/openms/include/OpenMS/FORMAT/DATAACCESS/SiriusFragmentAnnotation.h:
```diff
@@ -80,6 +80,11 @@ namespace OpenMS
+          /**
+          @brief Extract columnname and index based in SIRIUS entries
+          */
+          static std::map< std::string, Size > extract_columnname_to_columnindex(CsvFile& csvfile);
```
can this be const CsvFile& ?

timosachsenberg commented on src/openms/source/ANALYSIS/ID/SiriusExportAlgorithm.cpp:
```diff
@@ -0,0 +1,273 @@
+      os.precision(12);
+
+      std::vector<SiriusMSFile::CompoundInfo> v_cmpinfo; // To store compound information for all files
+      for (size_t i = 0; i < mzML_files.size(); ++i) {
```
the formatting seems to be a bit of in this file in general. e.g.
```suggestion
      for (size_t i = 0; i < mzML_files.size(); ++i) 
      {
```
or tab sizes

timosachsenberg commented on src/openms/source/ANALYSIS/ID/SiriusMSConverter.cpp:
```diff
@@ -111,15 +108,18 @@ namespace OpenMS
     // let SIRIUS sort it out using fragment annotation
     for (unsigned int k = 0; k != v_description.size(); ++k)
-    {
+    { 
```
```suggestion
    {
```

timosachsenberg commented on src/openms/source/ANALYSIS/ID/SiriusMSConverter.cpp:
```diff
@@ -702,17 +699,51 @@ namespace OpenMS
+      file << "cmp\tfile_index\tpmass\tpint_mono\trt\tfmz\tfid\tformula\tcharge\tionization\tdes\tspecref_format\tsource_file\tsource_format\tnative_ids_id\tm_ids_id\n";
+
+      // Iterate over the vector and write each object's attributes
+      for (const auto& info : v_cmpinfo) {
```
```suggestion
      for (const auto& info : v_cmpinfo)
      {
```

axelwalter committed:
Update CHANGELOG

Co-authored-by: Timo Sachsenberg <timo.sachsenberg@uni-tuebingen.de>

axelwalter committed:
Update src/openms/source/ANALYSIS/ID/SiriusMSConverter.cpp

Co-authored-by: Timo Sachsenberg <timo.sachsenberg@uni-tuebingen.de>

axelwalter committed:
Update src/openms/source/ANALYSIS/ID/SiriusMSConverter.cpp

Co-authored-by: Timo Sachsenberg <timo.sachsenberg@uni-tuebingen.de>

axelwalter committed:
update docs

axelwalter committed:
pass CsvFile as const reference

axelwalter committed:
fix formatting

timosachsenberg commented:
@CodiumAI-Agent /review

QodoAI-Agent commented:
## PR Analysis

### (review updated until commit https://github.com/OpenMS/OpenMS/commit/ebcaaaaa27c43b8efb86c095689528538d0975f6)


- 🎯 **Main theme:** Replacing SiriusAdapter with SiriusExport and adapting AssayGeneratorMetabo
- 📝 **PR summary:** This PR replaces the SiriusAdapter with SiriusExport and adapts AssayGeneratorMetabo to accept an existing SIRIUS workspace as input instead of calling SIRIUS during the workflow. It also includes updates to the documentation and tests.
- 📌 **Type of PR:** Enhancement
- 🧪 **Relevant tests added:** Yes
- ⏱️ **Estimated effort to review [1-5]:** 5, because the PR involves significant changes to the codebase, including the removal of several files and the addition of new ones. It requires a deep understanding of the SiriusAdapter, SiriusExport, and AssayGeneratorMetabo to review effectively.
## PR Feedback



<strong>💡 General suggestions:</strong> The PR is well-structured and includes meaningful changes. It would be helpful to include more context about why these changes are necessary and how they improve the codebase or functionality. Additionally, it would be beneficial to explain how the new workflow with SiriusExport and AssayGeneratorMetabo works in the PR description.


<details><summary> <strong>🤖 Code feedback:</strong></summary><hr><table><tr><td>relevant file</td><td>src/openms/source/ANALYSIS/ID/SiriusExportAlgorithm.cpp</td></tr><tr><td>suggestion &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</td><td><br>

**Consider using `std::any_of` instead of the for loop in the `hasFullNameParameter` function. This could potentially improve the readability and performance of the code. [medium]**
<br></td></tr><tr><td>relevant line</td><td><a href='https://github.com/OpenMS/OpenMS/pull/7234/files#diff-54ba82af6db850be2c3e356c8d4af04ee0d527e9e510702c670abe7d434d898dR102'>for (auto it = param_.begin(); it != param_.end(); ++it)</a></td></tr></table><hr><table><tr><td>relevant file</td><td>src/openms/source/ANALYSIS/ID/SiriusExportAlgorithm.cpp</td></tr><tr><td>suggestion &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</td><td><br>

**The `preprocessingSirius` function seems to be doing a lot of work. Consider breaking it down into smaller, more manageable functions. This could improve readability and maintainability of the code. [important]**
<br></td></tr><tr><td>relevant line</td><td><a href='https://github.com/OpenMS/OpenMS/pull/7234/files#diff-54ba82af6db850be2c3e356c8d4af04ee0d527e9e510702c670abe7d434d898dR116'>void SiriusExportAlgorithm::preprocessingSirius(const String& featureinfo,</a></td></tr></table><hr><table><tr><td>relevant file</td><td>src/openms/source/ANALYSIS/ID/SiriusExportAlgorithm.cpp</td></tr><tr><td>suggestion &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</td><td><br>

**The `logFeatureSpectraNumber` function could benefit from more descriptive logging messages. It would be helpful to provide more context about what the numbers being logged represent. [medium]**
<br></td></tr><tr><td>relevant line</td><td><a href='https://github.com/OpenMS/OpenMS/pull/7234/files#diff-54ba82af6db850be2c3e356c8d4af04ee0d527e9e510702c670abe7d434d898dR169'>void SiriusExportAlgorithm::logFeatureSpectraNumber(const String& featureinfo,</a></td></tr></table><hr><table><tr><td>relevant file</td><td>src/openms/source/ANALYSIS/ID/SiriusExportAlgorithm.cpp</td></tr><tr><td>suggestion &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</td><td><br>

**The `run` function is**
<br></td></tr><tr><td>relevant line</td><td></td></tr></table><hr></details>

<hr>

<details> <summary><strong>✨ Usage guide:</strong></summary><hr> 

**Overview:**
The `review` tool scans the PR code changes, and generates a PR review. The tool can be triggered [automatically](https://github.com/Codium-ai/pr-agent/blob/main/Usage.md#github-app-automatic-tools) every time a new PR is opened, or can be invoked manually by commenting on any PR.
When commenting, to edit [configurations](https://github.com/Codium-ai/pr-agent/blob/main/pr_agent/settings/configuration.toml#L19) related to the review tool (`pr_reviewer` section), use the following template:
```
/review --pr_reviewer.some_config1=... --pr_reviewer.some_config2=...
```
With a [configuration file](https://github.com/Codium-ai/pr-agent/blob/main/Usage.md#working-with-github-app), use the following template:
```
[pr_reviewer]
some_config1=...
some_config2=...
```
    

<table><tr><td><details> <summary><strong> Utilizing extra instructions</strong></summary><hr>

The `review` tool can be configured with extra instructions, which can be used to guide the model to a feedback tailored to the needs of your project.

Be specific, clear, and concise in the instructions. With extra instructions, you are the prompter. Specify the relevant sub-tool, and the relevant aspects of the PR that you want to emphasize.

Examples for extra instructions:
```
[pr_reviewer] # /review #
extra_instructions="""
In the code feedback section, emphasize the following:
- Does the code logic cover relevant edge cases?
- Is the code logic clear and easy to understand?
- Is the code logic efficient?
...
"""
```
Use triple quotes to write multi-line instructions. Use bullet points to make the instructions more readable.
        

</details></td></tr>

<tr><td><details> <summary><strong> How to enable\disable automation</strong></summary><hr>

- When you first install PR-Agent app, the [default mode](https://github.com/Codium-ai/pr-agent/blob/main/Usage.md#github-app-automatic-tools) for the `review` tool is:
```
pr_commands = ["/review", ...]
```
meaning the `review` tool will run automatically on every PR, with the default configuration.
Edit this field to enable/disable the tool, or to change the used configurations
        

</details></td></tr>

<tr><td><details> <summary><strong> About the 'Code feedback' section</strong></summary><hr>

The `review` tool provides several type of feedbacks, one of them is code suggestions.
If you are interested **only** in the code suggestions, it is recommended to use the [`improve`](https://github.com/Codium-ai/pr-agent/blob/main/docs/IMPROVE.md) feature instead, since it dedicated only to code suggestions, and usually gives better results.
Use the `review` tool if you want to get a more comprehensive feedback, which includes code suggestions as well.


</details></td></tr>

<tr><td><details> <summary><strong> Auto-labels</strong></summary><hr>

The `review` tool can auto-generate two specific types of labels for a PR:
- a `possible security issue` label, that detects possible [security issues](https://github.com/Codium-ai/pr-agent/blob/tr/user_description/pr_agent/settings/pr_reviewer_prompts.toml#L136) (`enable_review_labels_security` flag)
- a `Review effort [1-5]: x` label, where x is the estimated effort to review the PR (`enable_review_labels_effort` flag)


</details></td></tr>

<tr><td><details> <summary><strong> Extra sub-tools</strong></summary><hr>

The `review` tool provides a collection of possible feedbacks about a PR.
It is recommended to review the [possible options](https://github.com/Codium-ai/pr-agent/blob/main/docs/REVIEW.md#enabledisable-features), and choose the ones relevant for your use case.
Some of the feature that are disabled by default are quite useful, and should be considered for enabling. For example: 
`require_score_review`, `require_soc2_review`, `enable_review_labels_effort`, and more.


</details></td></tr>



<tr><td><details> <summary><strong> More PR-Agent commands</strong></summary><hr> 

> To invoke the PR-Agent, add a comment using one of the following commands:  
> - **/review**: Request a review of your Pull Request.   
> - **/describe**: Update the PR title and description based on the contents of the PR.   
> - **/improve [--extended]**: Suggest code improvements. Extended mode provides a higher quality feedback.   
> - **/ask \<QUESTION\>**: Ask a question about the PR.   
> - **/update_changelog**: Update the changelog based on the PR's contents.   
> - **/add_docs** 💎: Generate docstring for new components introduced in the PR.   
> - **/generate_labels** 💎: Generate labels for the PR based on the PR's contents.   
> - **/analyze** 💎: Automatically analyzes the PR, and presents changes walkthrough for each component.   

>See the [tools guide](https://github.com/Codium-ai/pr-agent/blob/main/docs/TOOLS_GUIDE.md) for more details.
>To list the possible configuration parameters, add a **/config** comment.   
 


</details></td></tr>

</table>

See the [review usage](https://github.com/Codium-ai/pr-agent/blob/main/docs/REVIEW.md) page for a comprehensive guide on using this tool.


</details>

QodoAI-Agent commented on src/openms/source/ANALYSIS/ID/SiriusExportAlgorithm.cpp:
```diff
@@ -0,0 +1,274 @@
+    // ################
+    // Algorithm
+    // ################
+    void SiriusExportAlgorithm::preprocessingSirius(const String& featureinfo,
```
Consider refactoring the `preprocessingSirius` method to reduce its complexity and improve readability. This method is quite long and does several things, which could potentially be broken down into smaller, more manageable methods. [important]

QodoAI-Agent commented on src/openms/source/ANALYSIS/ID/SiriusExportAlgorithm.cpp:
```diff
@@ -0,0 +1,274 @@
+    // ####################
+    // Parameter handling
+    // ####################
+    SiriusExportAlgorithm::ParameterModifier SiriusExportAlgorithm::ParameterSection::parameter(
```
It would be beneficial to add more detailed comments explaining the purpose and functionality of the `ParameterSection` and `ParameterModifier` classes. While the code is generally well-commented, these classes could benefit from additional context. [medium]

QodoAI-Agent commented on src/openms/source/ANALYSIS/ID/SiriusExportAlgorithm.cpp:
```diff
@@ -0,0 +1,274 @@
+    // ################
+    void SiriusExportAlgorithm::preprocessingSirius(const String& featureinfo,
+                                                     const MSExperiment& spectra,
+                                                     FeatureMapping::FeatureMappingInfo& fm_info,
```
Consider using more descriptive variable names. For example, `fm_info` could be renamed to something more descriptive like `feature_mapping_info`. This would improve code readability and maintainability. [medium]

QodoAI-Agent commented:
**[Persistent review](https://github.com/OpenMS/OpenMS/pull/7234#issuecomment-1895881235)** updated to latest commit https://github.com/OpenMS/OpenMS/commit/ebcaaaaa27c43b8efb86c095689528538d0975f6

axelwalter committed:
use default param handler for SiriusExportAlgorithm

axelwalter committed:
descriptive parameter names

axelwalter committed:
set min int for mass trace filter parameter

axelwalter committed:
resolve merge conflicts

axelwalter committed:
update docs

axelwalter committed:
fix merge issues

axelwalter committed:
add AccurateMassSearch to executables.cmake
- was accidentally removed

timosachsenberg commented:
/review

timosachsenberg commented:
- needs follow up PR to remove sirius executables, and also remove from installer scripts and cmake files

oliveralka commented on src/openms/source/ANALYSIS/ID/SiriusExportAlgorithm.cpp:
```diff
@@ -0,0 +1,180 @@
+      // returns Log of feature and/or spectra number
+      SiriusExportAlgorithm::logFeatureSpectraNumber(feature_info_to_pass, feature_ms2_indices, spectra);
+
+      // bool no_masstrace_info_isotope_pattern = (no_masstrace_info_isotope_pattern_ == "true");
```
This can be removed. Should be handled in "isNoMasstraceInfoIsotopePattern()".

oliveralka commented on src/openms/source/ANALYSIS/TARGETED/MetaboTargetedTargetDecoy.cpp:
```diff
@@ -74,68 +74,65 @@ namespace OpenMS
   }
 
-  void MetaboTargetedTargetDecoy::resolveOverlappingTargetDecoyMassesByIndividualMassShift(TargetedExperiment& t_exp, std::vector<MetaboTargetedTargetDecoy::MetaboTargetDecoyMassMapping>& mappings, const double& mass_to_add)
+  void MetaboTargetedTargetDecoy::resolveOverlappingTargetDecoyMassesByDecoyMassShift(TargetedExperiment& t_exp, std::vector<MetaboTargetedTargetDecoy::MetaboTargetDecoyMassMapping>& mappings, const double& mass_to_add, const double& mz_tol, const String& mz_tol_unit)
```
As far as I remember, this was introduced for cases were the target mass-to-charge was exactly the decoy mass-to-charge. This happended sometimes if the re-rooting did not work properly. That said it IMHO does not hurt to have a tolerance here.

axelwalter replied:
The masses can be exactly the same (if `use_exact_mass` is true) or slightly different if false. In that case we have experimental m/z for the targets and exact m/z for the decoys.

oliveralka replied:
Ok, thanks for the clarifiation!

oliveralka commented on src/openms/source/ANALYSIS/TARGETED/MetaboTargetedTargetDecoy.cpp:
```diff
@@ -74,68 +74,65 @@ namespace OpenMS
+                    std::abs(decoy_mz - target_mz);
+
+                // If the difference is within the specified tolerance, add the decoy m/z to the matched set.
+                if (difference <= mz_tol) {
```
If the difference is smaller than the mz_tol the masses are too similar. Is my understanding correct here? Might be worth to add another small comment, but is not a must.

axelwalter replied:
Exactly, added another comment to clarify.

oliveralka commented on src/tests/class_tests/openms/source/SiriusFragmentAnnotation_test.cpp:
```diff
@@ -91,22 +91,19 @@ START_SECTION(static void extractSiriusDecoyAnnotationMapping(const String& path
     MSSpectrum decoy_msspectrum = SiriusFragmentAnnotation::extractAnnotationsFromSiriusFile(test_path, 1, true, false)[0];
 
-    TEST_STRING_SIMILAR(decoy_msspectrum.getNativeID(), "sample=1 period=1 cycle=657 experiment=5|sample=1 period=1 cycle=658 experiment=6|sample=1 period=1 cycle=659 experiment=7");
+    TEST_STRING_SIMILAR(decoy_msspectrum.getNativeID(), "sample=1 period=1 cycle=676 experiment=4|sample=1 period=1 cycle=677 experiment=5|sample=1 period=1 cycle=678 experiment=3");
```
It seems you have found some nicer test data ;)

oliveralka commented on src/topp/SiriusExport.cpp:
```diff
@@ -0,0 +1,137 @@
+
+  @brief De novo metabolite identification.
+
+  Export input file SIRIUS, a tool for metabolomics data analysis with several subtools, each serving specific purposes:
```
```suggestion
  Exports an input file for SIRIUS, a tool for metabolomics data analysis with several subtools, each serving specific purposes:
```

oliveralka commented on src/topp/SiriusExport.cpp:
```diff
@@ -0,0 +1,137 @@
+  - CANOPUS: Predict compound categories for each compound individually based on its predicted molecular fingerprint (CSI:FingerID) using CANOPUS.
+  - Passatutto: Compute decoy spectra based  on the fragmentation trees of the given input spectra. If no molecular formula is provided in the input, the top scoring computed formula is used. Required to include decoys in an OpenSwathWorkflow assay library generated by the AssayGeneratorMetabo TOPP tool.
+
+  Sirius can be found on https://bio.informatik.uni-jena.de/software/sirius/
```
```suggestion
  Sirius can be found at https://bio.informatik.uni-jena.de/software/sirius/
```

oliveralka reviewed (commented):
Overview: SiriusAdapter was removed; All mztab export functionality was removed; SiriusAdapterAlgorithm was restructured to allow the export of an input file for SIRIUS. AssayGeneratorMetabo was seperated into a heuristic and a Sirius version (working by using the generated SIRIUS project/workspace).

All in all a very nice change to reduce the dependency to ever-changing THIRDPARTY modules. 

There are two things I am a bit worried about. The first is that, even though very nice class tests were introduced, there are no more integration tests using SIRIUS. So the only way we know something broke (e.g., due to updates in SIRIUS) is based on either manual testing or user feedback. 

The other concern is that this major breaking change will break the current workflows which are dependent on the SiriusAdapterAlgorithms (e.g., DIAMetAlyzer) or its python wrapper (e.g., https://github.com/eeko-kon/pyOpenMS_UmetaFlow/blob/master/5_SIRIUS_CSI.ipynb). Basically these will not be natively supported anymore with newer OpenMS versions. I think this should be transparently communicated and discussed.

oliveralka commented:
> * needs follow up PR to remove sirius executables, and also remove from installer scripts and cmake files

I guess the tutorials and workflow(s) need to be updated as well?

axelwalter committed:
Update src/topp/SiriusExport.cpp

Co-authored-by: Oliver Alka <oliver.alka@gmx.de>

axelwalter committed:
Update src/topp/SiriusExport.cpp

Co-authored-by: Oliver Alka <oliver.alka@gmx.de>

axelwalter committed:
update comments

axelwalter committed:
Merge branch 'develop' into sirius-export

axelwalter commented:
@oliveralka thanks a lot for the review! The testing of future SIRIUS versions is indeed a concern. For now I have made a comment in the AssayGeneratorMetaboSirius docs which SIRIUS versions are tested.
The current DIAMetAlyzer and UmetaFlow workflows will need updates or fixed OpenMS 3.1 versions.

axelwalter commented:
Once merged, I will make PRs with the required follow-up changes.

oliveralka reviewed (approved):
FYI: @timosachsenberg @axelwalter 

It looks good time, if all tests passes it seems ready to go. I hope it is Ok, that I approve the PR. I am not sure how this is currently handled or if the "review-bot" should work its magic again before the merge.

axelwalter committed:
Merge branch 'develop' into sirius-export
